Run mysql under mysql user if ddev-dbserver container is run as root

hussainweb opened the pull request:
## The Problem/Issue/Bug:

The problem is detailed in #3025.

## How this PR Solves The Problem:

It solves it according to the solution described in the above issue. This is the same way the official images do this as well.

## Manual Testing Instructions:
WIP

## Automated Testing Overview:
WIP

## Related Issue Link(s):
#3025 

## Release/Deployment notes:
This shouldn't affect any other workflows. New database docker images would be needed as part of the release.

<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/3026"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/drud/ddev?pullRequest=3026) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/drud/ddev?pullRequest=3026) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/drud/ddev?pullRequest=3026) it.</sub>

rfay reviewed (changes requested):
I don't expect that this will work, because the container will run as the mysql user I guess, which is 999 in this container. But the /var/lib/mysql will already have been initialized with another uid, right? 

I think it might be better to allow mysql to run as root (if that works for both mysql and mariadb). 

You'll need to actually manually test this by building it (`make VERSION=<your_tag>` in containers/ddev-dbserver), and you'll need to have a specific manual test plan.

There are some hints about how to build and use a PR that has a particular container at https://github.com/drud/ddev/blob/master/docs/developers/building-contributing.md

Please try that out manually and see how you do. 

Again, it sounds to me like this might not be the most obvious answer to the problem you're trying to solve.

hussainweb commented:
Thanks, @rfay for reviewing this. I am considering your message and will build and test this manually. But for the sake of clarity, I will try to restate the problem here.

I want to be able to run this image directly similar to how we can already run other images from MySQL and MariaDB. Consider these hypothetical examples. Also, you can see the same message about switching to the `mysql` user in logs for both official images. Logs below.

Again, thanks for reviewing this and I will get back to you with the results from my testing.

---

MySQL 8.0 with emulation (as I have M1)
`docker run --platform linux/amd64 -e MYSQL_ALLOW_EMPTY_PASSWORD=true mysql:8.0`

<details><summary>Full log</summary>
<p>

```
2021-05-31 18:50:17+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.25-1debian10 started.
2021-05-31 18:50:17+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2021-05-31 18:50:17+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.25-1debian10 started.
2021-05-31 18:50:18+00:00 [Note] [Entrypoint]: Initializing database files
2021-05-31T18:50:18.361203Z 0 [System] [MY-013169] [Server] /usr/sbin/mysqld (mysqld 8.0.25) initializing of server in progress as process 100
2021-05-31T18:50:18.435033Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2021-05-31T18:50:18.454483Z 1 [ERROR] [MY-012585] [InnoDB] Linux Native AIO interface is not supported on this platform. Please check your OS documentation and install appropriate binary of InnoDB.
2021-05-31T18:50:18.455114Z 1 [Warning] [MY-012654] [InnoDB] Linux Native AIO disabled.
2021-05-31T18:50:19.200329Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2021-05-31T18:50:22.905901Z 6 [Warning] [MY-010453] [Server] root@localhost is created with an empty password ! Please consider switching off the --initialize-insecure option.
2021-05-31 18:50:28+00:00 [Note] [Entrypoint]: Database files initialized
2021-05-31 18:50:28+00:00 [Note] [Entrypoint]: Starting temporary server
2021-05-31T18:50:29.156527Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.25) starting as process 151
2021-05-31T18:50:29.303277Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2021-05-31T18:50:29.337274Z 1 [ERROR] [MY-012585] [InnoDB] Linux Native AIO interface is not supported on this platform. Please check your OS documentation and install appropriate binary of InnoDB.
2021-05-31T18:50:29.338086Z 1 [Warning] [MY-012654] [InnoDB] Linux Native AIO disabled.
2021-05-31T18:50:29.722859Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2021-05-31T18:50:30.269302Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Socket: /var/run/mysqld/mysqlx.sock
2021-05-31T18:50:30.828073Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2021-05-31T18:50:30.829604Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
2021-05-31T18:50:30.836418Z 0 [Warning] [MY-011810] [Server] Insecure configuration for --pid-file: Location '/var/run/mysqld' in the path is accessible to all OS users. Consider choosing a different directory.
2021-05-31T18:50:30.961600Z 0 [System] [MY-010931] [Server] /usr/sbin/mysqld: ready for connections. Version: '8.0.25'  socket: '/var/run/mysqld/mysqld.sock'  port: 0  MySQL Community Server - GPL.
2021-05-31 18:50:31+00:00 [Note] [Entrypoint]: Temporary server started.
Warning: Unable to load '/usr/share/zoneinfo/iso3166.tab' as time zone. Skipping it.
Warning: Unable to load '/usr/share/zoneinfo/leap-seconds.list' as time zone. Skipping it.
Warning: Unable to load '/usr/share/zoneinfo/zone.tab' as time zone. Skipping it.
Warning: Unable to load '/usr/share/zoneinfo/zone1970.tab' as time zone. Skipping it.

2021-05-31 18:50:40+00:00 [Note] [Entrypoint]: Stopping temporary server
2021-05-31T18:50:40.723314Z 10 [System] [MY-013172] [Server] Received SHUTDOWN from user root. Shutting down mysqld (Version: 8.0.25).
2021-05-31T18:50:42.325430Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.25)  MySQL Community Server - GPL.
2021-05-31 18:50:42+00:00 [Note] [Entrypoint]: Temporary server stopped

2021-05-31 18:50:42+00:00 [Note] [Entrypoint]: MySQL init process done. Ready for start up.

2021-05-31T18:50:43.628774Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.25) starting as process 1
2021-05-31T18:50:43.736098Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2021-05-31T18:50:43.767719Z 1 [ERROR] [MY-012585] [InnoDB] Linux Native AIO interface is not supported on this platform. Please check your OS documentation and install appropriate binary of InnoDB.
2021-05-31T18:50:43.768442Z 1 [Warning] [MY-012654] [InnoDB] Linux Native AIO disabled.
2021-05-31T18:50:44.101011Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2021-05-31T18:50:44.719014Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Bind-address: '::' port: 33060, socket: /var/run/mysqld/mysqlx.sock
2021-05-31T18:50:44.984187Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2021-05-31T18:50:44.985481Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
2021-05-31T18:50:44.995513Z 0 [Warning] [MY-011810] [Server] Insecure configuration for --pid-file: Location '/var/run/mysqld' in the path is accessible to all OS users. Consider choosing a different directory.
2021-05-31T18:50:45.104918Z 0 [System] [MY-010931] [Server] /usr/sbin/mysqld: ready for connections. Version: '8.0.25'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  MySQL Community Server - GPL.
2021-05-31T18:50:54.730472Z 0 [System] [MY-013172] [Server] Received SHUTDOWN from user <via user signal>. Shutting down mysqld (Version: 8.0.25).
2021-05-31T18:50:55.826303Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.25)  MySQL Community Server - GPL.
```

</p>
</details>

---

MariaDB 10.5
`docker run -e MARIADB_ALLOW_EMPTY_ROOT_PASSWORD=true mariadb:10.5`

<details><summary>Full log</summary>
<p>

```
2021-05-31 18:52:33+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.5.10+maria~focal started.
2021-05-31 18:52:33+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2021-05-31 18:52:33+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.5.10+maria~focal started.
2021-05-31 18:52:33+00:00 [Note] [Entrypoint]: Initializing database files


PLEASE REMEMBER TO SET A PASSWORD FOR THE MariaDB root USER !
To do so, start the server, then issue the following command:

'/usr/bin/mysql_secure_installation'

which will also give you the option of removing the test
databases and anonymous user created by default.  This is
strongly recommended for production servers.

See the MariaDB Knowledgebase at https://mariadb.com/kb or the
MySQL manual for more instructions.

Please report any problems at https://mariadb.org/jira

The latest information about MariaDB is available at https://mariadb.org/.
You can find additional information about the MySQL part at:
https://dev.mysql.com
Consider joining MariaDB's strong and vibrant community:
https://mariadb.org/get-involved/

2021-05-31 18:52:33+00:00 [Note] [Entrypoint]: Database files initialized
2021-05-31 18:52:33+00:00 [Note] [Entrypoint]: Starting temporary server
2021-05-31 18:52:33+00:00 [Note] [Entrypoint]: Waiting for server startup
2021-05-31 18:52:33 0 [Note] mysqld (mysqld 10.5.10-MariaDB-1:10.5.10+maria~focal) starting as process 103 ...
2021-05-31 18:52:33 0 [Note] InnoDB: Uses event mutexes
2021-05-31 18:52:33 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2021-05-31 18:52:33 0 [Note] InnoDB: Number of pools: 1
2021-05-31 18:52:33 0 [Note] InnoDB: Using ARMv8 crc32 + pmull instructions
2021-05-31 18:52:33 0 [Note] mysqld: O_TMPFILE is not supported on /tmp (disabling future attempts)
2021-05-31 18:52:33 0 [Note] InnoDB: Using Linux native AIO
2021-05-31 18:52:33 0 [Note] InnoDB: Initializing buffer pool, total size = 134217728, chunk size = 134217728
2021-05-31 18:52:33 0 [Note] InnoDB: Completed initialization of buffer pool
2021-05-31 18:52:33 0 [Note] InnoDB: 128 rollback segments are active.
2021-05-31 18:52:33 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2021-05-31 18:52:33 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2021-05-31 18:52:33 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2021-05-31 18:52:33 0 [Note] InnoDB: 10.5.10 started; log sequence number 45118; transaction id 20
2021-05-31 18:52:33 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
2021-05-31 18:52:33 0 [Note] Plugin 'FEEDBACK' is disabled.
2021-05-31 18:52:33 0 [Note] InnoDB: Buffer pool(s) load completed at 210531 18:52:33
2021-05-31 18:52:33 0 [Warning] 'user' entry 'root@bba28fe4896c' ignored in --skip-name-resolve mode.
2021-05-31 18:52:33 0 [Warning] 'proxies_priv' entry '@% root@bba28fe4896c' ignored in --skip-name-resolve mode.
2021-05-31 18:52:33 0 [Note] Reading of all Master_info entries succeeded
2021-05-31 18:52:33 0 [Note] Added new Master_info '' to hash table
2021-05-31 18:52:33 0 [Note] mysqld: ready for connections.
Version: '10.5.10-MariaDB-1:10.5.10+maria~focal'  socket: '/run/mysqld/mysqld.sock'  port: 0  mariadb.org binary distribution
2021-05-31 18:52:34+00:00 [Note] [Entrypoint]: Temporary server started.
Warning: Unable to load '/usr/share/zoneinfo/leap-seconds.list' as time zone. Skipping it.
Warning: Unable to load '/usr/share/zoneinfo/leapseconds' as time zone. Skipping it.
Warning: Unable to load '/usr/share/zoneinfo/tzdata.zi' as time zone. Skipping it.
2021-05-31 18:52:36 5 [Warning] 'proxies_priv' entry '@% root@bba28fe4896c' ignored in --skip-name-resolve mode.

2021-05-31 18:52:36+00:00 [Note] [Entrypoint]: Stopping temporary server
2021-05-31 18:52:36 0 [Note] mysqld (initiated by: root[root] @ localhost []): Normal shutdown
2021-05-31 18:52:36 0 [Note] Event Scheduler: Purging the queue. 0 events
2021-05-31 18:52:36 0 [Note] InnoDB: FTS optimize thread exiting.
2021-05-31 18:52:36 0 [Note] InnoDB: Starting shutdown...
2021-05-31 18:52:36 0 [Note] InnoDB: Dumping buffer pool(s) to /var/lib/mysql/ib_buffer_pool
2021-05-31 18:52:36 0 [Note] InnoDB: Buffer pool(s) dump completed at 210531 18:52:36
2021-05-31 18:52:36 0 [Note] InnoDB: Removed temporary tablespace data file: "ibtmp1"
2021-05-31 18:52:36 0 [Note] InnoDB: Shutdown completed; log sequence number 45130; transaction id 21
2021-05-31 18:52:36 0 [Note] mysqld: Shutdown complete

2021-05-31 18:52:37+00:00 [Note] [Entrypoint]: Temporary server stopped

2021-05-31 18:52:37+00:00 [Note] [Entrypoint]: MariaDB init process done. Ready for start up.

2021-05-31 18:52:37 0 [Note] mysqld (mysqld 10.5.10-MariaDB-1:10.5.10+maria~focal) starting as process 1 ...
2021-05-31 18:52:37 0 [Note] InnoDB: Uses event mutexes
2021-05-31 18:52:37 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2021-05-31 18:52:37 0 [Note] InnoDB: Number of pools: 1
2021-05-31 18:52:37 0 [Note] InnoDB: Using ARMv8 crc32 + pmull instructions
2021-05-31 18:52:37 0 [Note] mysqld: O_TMPFILE is not supported on /tmp (disabling future attempts)
2021-05-31 18:52:37 0 [Note] InnoDB: Using Linux native AIO
2021-05-31 18:52:37 0 [Note] InnoDB: Initializing buffer pool, total size = 134217728, chunk size = 134217728
2021-05-31 18:52:37 0 [Note] InnoDB: Completed initialization of buffer pool
2021-05-31 18:52:37 0 [Note] InnoDB: 128 rollback segments are active.
2021-05-31 18:52:37 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2021-05-31 18:52:37 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2021-05-31 18:52:37 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2021-05-31 18:52:37 0 [Note] InnoDB: 10.5.10 started; log sequence number 45130; transaction id 20
2021-05-31 18:52:37 0 [Note] Plugin 'FEEDBACK' is disabled.
2021-05-31 18:52:37 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
2021-05-31 18:52:37 0 [Note] InnoDB: Buffer pool(s) load completed at 210531 18:52:37
2021-05-31 18:52:37 0 [Note] Server socket created on IP: '::'.
2021-05-31 18:52:37 0 [Warning] 'proxies_priv' entry '@% root@bba28fe4896c' ignored in --skip-name-resolve mode.
2021-05-31 18:52:37 0 [Note] Reading of all Master_info entries succeeded
2021-05-31 18:52:37 0 [Note] Added new Master_info '' to hash table
2021-05-31 18:52:37 0 [Note] mysqld: ready for connections.
Version: '10.5.10-MariaDB-1:10.5.10+maria~focal'  socket: '/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
2021-05-31 18:52:59 0 [Note] mysqld (initiated by: unknown): Normal shutdown
2021-05-31 18:52:59 0 [Note] Event Scheduler: Purging the queue. 0 events
2021-05-31 18:52:59 0 [Note] InnoDB: FTS optimize thread exiting.
2021-05-31 18:52:59 0 [Note] InnoDB: Starting shutdown...
2021-05-31 18:52:59 0 [Note] InnoDB: Dumping buffer pool(s) to /var/lib/mysql/ib_buffer_pool
2021-05-31 18:52:59 0 [Note] InnoDB: Buffer pool(s) dump completed at 210531 18:52:59
2021-05-31 18:52:59 0 [Note] InnoDB: Removed temporary tablespace data file: "ibtmp1"
2021-05-31 18:52:59 0 [Note] InnoDB: Shutdown completed; log sequence number 45142; transaction id 21
2021-05-31 18:52:59 0 [Note] mysqld: Shutdown complete
```
</p>
</details>

---

Bitnami MariaDB 10.5 image
`docker run -e ALLOW_EMPTY_PASSWORD=yes bitnami/mariadb:10.5`

<details><summary>Full log</summary>
<p>

```
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
[38;5;6mmariadb [38;5;5m18:55:22.06 [0m
[38;5;6mmariadb [38;5;5m18:55:22.08 [0m[1mWelcome to the Bitnami mariadb container[0m
[38;5;6mmariadb [38;5;5m18:55:22.11 [0mSubscribe to project updates by watching [1mhttps://github.com/bitnami/bitnami-docker-mariadb[0m
[38;5;6mmariadb [38;5;5m18:55:22.12 [0mSubmit issues and feature requests at [1mhttps://github.com/bitnami/bitnami-docker-mariadb/issues[0m
[38;5;6mmariadb [38;5;5m18:55:22.14 [0m
[38;5;6mmariadb [38;5;5m18:55:22.16 [0m[38;5;2mINFO [0m ==> ** Starting MariaDB setup **
[38;5;6mmariadb [38;5;5m18:55:22.29 [0m[38;5;2mINFO [0m ==> Validating settings in MYSQL_*/MARIADB_* env vars
[38;5;6mmariadb [38;5;5m18:55:22.31 [0m[38;5;3mWARN [0m ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
[38;5;6mmariadb [38;5;5m18:55:22.35 [0m[38;5;2mINFO [0m ==> Initializing mariadb database
[38;5;6mmariadb [38;5;5m18:55:22.49 [0m[38;5;2mINFO [0m ==> Updating 'my.cnf' with custom configuration
[38;5;6mmariadb [38;5;5m18:55:22.54 [0m[38;5;2mINFO [0m ==> Installing database
[38;5;6mmariadb [38;5;5m18:55:24.02 [0m[38;5;2mINFO [0m ==> Starting mariadb in background
[38;5;6mmariadb [38;5;5m18:55:26.17 [0m[38;5;2mINFO [0m ==> Configuring authentication
[38;5;6mmariadb [38;5;5m18:55:26.59 [0m[38;5;2mINFO [0m ==> Running mysql_upgrade
[38;5;6mmariadb [38;5;5m18:55:28.65 [0m[38;5;2mINFO [0m ==> Stopping mariadb
[38;5;6mmariadb [38;5;5m18:55:31.46 [0m[38;5;2mINFO [0m ==> ** MariaDB setup finished! **

[38;5;6mmariadb [38;5;5m18:55:31.61 [0m[38;5;2mINFO [0m ==> ** Starting MariaDB **
2021-05-31 18:55:31 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (mysqld 10.5.10-MariaDB) starting as process 1 ...
2021-05-31 18:55:31 0 [Note] InnoDB: Uses event mutexes
2021-05-31 18:55:31 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2021-05-31 18:55:31 0 [Note] InnoDB: Number of pools: 1
2021-05-31 18:55:31 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2021-05-31 18:55:31 0 [Note] mysqld: O_TMPFILE is not supported on /opt/bitnami/mariadb/tmp (disabling future attempts)
2021-05-31 18:55:31 0 [Warning] InnoDB: Linux Native AIO disabled.
2021-05-31 18:55:31 0 [Note] InnoDB: Initializing buffer pool, total size = 134217728, chunk size = 134217728
2021-05-31 18:55:31 0 [Note] InnoDB: Completed initialization of buffer pool
2021-05-31 18:55:31 0 [Note] InnoDB: 128 rollback segments are active.
2021-05-31 18:55:31 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2021-05-31 18:55:31 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2021-05-31 18:55:31 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2021-05-31 18:55:31 0 [Note] InnoDB: 10.5.10 started; log sequence number 45142; transaction id 20
2021-05-31 18:55:31 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
2021-05-31 18:55:31 0 [Note] Plugin 'FEEDBACK' is disabled.
2021-05-31 18:55:31 0 [Note] InnoDB: Buffer pool(s) load completed at 210531 18:55:31
2021-05-31 18:55:31 0 [Note] Server socket created on IP: '0.0.0.0'.
2021-05-31 18:55:31 0 [Warning] 'proxies_priv' entry '@% root@8061afdfa581' ignored in --skip-name-resolve mode.
2021-05-31 18:55:31 0 [Note] Reading of all Master_info entries succeeded
2021-05-31 18:55:32 0 [Note] Added new Master_info '' to hash table
2021-05-31 18:55:32 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: ready for connections.
Version: '10.5.10-MariaDB'  socket: '/opt/bitnami/mariadb/tmp/mysql.sock'  port: 3306  Source distribution
2021-05-31 18:55:42 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (initiated by: unknown): Normal shutdown
2021-05-31 18:55:42 0 [Note] Event Scheduler: Purging the queue. 0 events
2021-05-31 18:55:42 0 [Note] InnoDB: FTS optimize thread exiting.
2021-05-31 18:55:42 0 [Note] InnoDB: Starting shutdown...
2021-05-31 18:55:42 0 [Note] InnoDB: Dumping buffer pool(s) to /bitnami/mariadb/data/ib_buffer_pool
2021-05-31 18:55:42 0 [Note] InnoDB: Buffer pool(s) dump completed at 210531 18:55:42
2021-05-31 18:55:42 0 [Note] InnoDB: Removed temporary tablespace data file: "ibtmp1"
2021-05-31 18:55:42 0 [Note] InnoDB: Shutdown completed; log sequence number 45154; transaction id 21
2021-05-31 18:55:42 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: Shutdown complete
```
</p>
</details>

---

DDEV image by running as mysql user
`docker run --user 999:999 drud/ddev-dbserver-mariadb-10.5:v1.17.0`
Error:
```
+ cp -r /home/.my.cnf /home/.bashrc /home/mysql/
cp: target '/home/mysql/' is not a directory
```

<details><summary>Full log</summary>
<p>

```
+ set -eu
+ set -o pipefail
+ SOCKET=/var/tmp/mysql.sock
+ rm -f /tmp/healthy
+ '[' 0 = 2 -a '' = restore_snapshot ']'
++ PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/sbin:/usr/local/bin:/usr/local/mysql/bin
++ mysqld -V
++ awk '{sub( /\.[0-9]+(-.*)?$/, "", $3); print $3 }'
+ server_db_version=10.5
++ echo '/mnt/ddev_config/mysql/*.cnf'
+ '[' -d /mnt/ddev_config/mysql -a '/mnt/ddev_config/mysql/*.cnf' '!=' '/mnt/ddev_config/mysql/*.cnf' ']'
+ export BACKUPTOOL=mariabackup
+ BACKUPTOOL=mariabackup
+ command -v xtrabackup
+ '[' '!' -f /var/lib/mysql/db_mariadb_version.txt ']'
+ '[' -z '' ']'
+ touch /tmp/initializing
+ target=/mysqlbase/
++ basename /mysqlbase/
+ name=mysqlbase
+ rm -rf '/var/lib/mysql/*' '/var/lib/mysql/.[a-z]*'
+ tee /var/log/mariabackup_prepare_mysqlbase.log
+ mariabackup --prepare --skip-innodb-use-native-aio --target-dir /mysqlbase/ --user=root --password=root --socket=/var/tmp/mysql.sock
Warning: World-writable config file '/mysqlbase/backup-my.cnf' is ignored
Warning: World-writable config file '/mysqlbase/backup-my.cnf' is ignored
Warning: World-writable config file '/mysqlbase/backup-my.cnf' is ignored
mariabackup based on MariaDB server 10.5.9-MariaDB debian-linux-gnu (aarch64)
[00] 2021-05-31 18:46:04 cd to /mysqlbase/
[00] 2021-05-31 18:46:04 open files limit requested 0, set to 1048576
[00] 2021-05-31 18:46:04 This target seems to be not prepared yet.
[00] 2021-05-31 18:46:04 mariabackup: using the following InnoDB configuration for recovery:
[00] 2021-05-31 18:46:04 innodb_data_home_dir = .
[00] 2021-05-31 18:46:04 innodb_data_file_path = ibdata1:10M:autoextend
[00] 2021-05-31 18:46:04 innodb_log_group_home_dir = .
[00] 2021-05-31 18:46:04 Starting InnoDB instance for recovery.
[00] 2021-05-31 18:46:04 mariabackup: Using 104857600 bytes for buffer pool (set by --use-memory parameter)
2021-05-31 18:46:04 0 [Note] InnoDB: Uses event mutexes
2021-05-31 18:46:04 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2021-05-31 18:46:04 0 [Note] InnoDB: Number of pools: 1
2021-05-31 18:46:04 0 [Note] InnoDB: Using ARMv8 crc32 + pmull instructions
2021-05-31 18:46:04 0 [Note] InnoDB: Initializing buffer pool, total size = 104857600, chunk size = 104857600
2021-05-31 18:46:04 0 [Note] InnoDB: Completed initialization of buffer pool
2021-05-31 18:46:04 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
2021-05-31 18:46:04 0 [Note] InnoDB: Starting crash recovery from checkpoint LSN=45130,45130
[00] 2021-05-31 18:46:04 Last binlog file , position 0
[00] 2021-05-31 18:46:04 completed OK!
+ mariabackup --copy-back --skip-innodb-use-native-aio --force-non-empty-directories --target-dir /mysqlbase/ --user=root --password=root --socket=/var/tmp/mysql.sock
+ tee /var/log/mariabackup_copy_back_mysqlbase.log
mariabackup based on MariaDB server 10.5.9-MariaDB debian-linux-gnu (aarch64)
[01] 2021-05-31 18:46:04 Copying ibdata1 to /var/lib/mysql/ibdata1
[01] 2021-05-31 18:46:04         ...done
[01] 2021-05-31 18:46:04 Copying ./db_mariadb_version.txt to /var/lib/mysql/db_mariadb_version.txt
[01] 2021-05-31 18:46:04         ...done
[01] 2021-05-31 18:46:04 Copying ./test/db.opt to /var/lib/mysql/test/db.opt
[01] 2021-05-31 18:46:04         ...done
[01] 2021-05-31 18:46:04 Copying ./mysql/table_stats.MAI to /var/lib/mysql/mysql/table_stats.MAI
[01] 2021-05-31 18:46:04         ...done
[01] 2021-05-31 18:46:04 Copying ./mysql/tables_priv.frm to /var/lib/mysql/mysql/tables_priv.frm
[01] 2021-05-31 18:46:04         ...done
[01] 2021-05-31 18:46:04 Copying ./mysql/slow_log.CSV to /var/lib/mysql/mysql/slow_log.CSV
[01] 2021-05-31 18:46:04         ...done
[01] 2021-05-31 18:46:04 Copying ./mysql/db.MAD to /var/lib/mysql/mysql/db.MAD
[01] 2021-05-31 18:46:04         ...done
[01] 2021-05-31 18:46:04 Copying ./mysql/slow_log.frm to /var/lib/mysql/mysql/slow_log.frm
[01] 2021-05-31 18:46:04         ...done
[01] 2021-05-31 18:46:04 Copying ./mysql/time_zone_transition.frm to /var/lib/mysql/mysql/time_zone_transition.frm
[01] 2021-05-31 18:46:04         ...done
[01] 2021-05-31 18:46:04 Copying ./mysql/proc.MAD to /var/lib/mysql/mysql/proc.MAD
[01] 2021-05-31 18:46:04         ...done
[01] 2021-05-31 18:46:04 Copying ./mysql/time_zone_transition.MAD to /var/lib/mysql/mysql/time_zone_transition.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/help_relation.frm to /var/lib/mysql/mysql/help_relation.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/tables_priv.MAI to /var/lib/mysql/mysql/tables_priv.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/column_stats.frm to /var/lib/mysql/mysql/column_stats.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/time_zone_transition_type.frm to /var/lib/mysql/mysql/time_zone_transition_type.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/help_topic.frm to /var/lib/mysql/mysql/help_topic.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/index_stats.MAD to /var/lib/mysql/mysql/index_stats.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/func.MAI to /var/lib/mysql/mysql/func.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/transaction_registry.ibd to /var/lib/mysql/mysql/transaction_registry.ibd
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/columns_priv.MAD to /var/lib/mysql/mysql/columns_priv.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/innodb_index_stats.frm to /var/lib/mysql/mysql/innodb_index_stats.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/transaction_registry.frm to /var/lib/mysql/mysql/transaction_registry.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/time_zone_name.MAI to /var/lib/mysql/mysql/time_zone_name.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/index_stats.MAI to /var/lib/mysql/mysql/index_stats.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/event.MAD to /var/lib/mysql/mysql/event.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/gtid_slave_pos.ibd to /var/lib/mysql/mysql/gtid_slave_pos.ibd
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/plugin.MAD to /var/lib/mysql/mysql/plugin.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/proc.frm to /var/lib/mysql/mysql/proc.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/time_zone.MAD to /var/lib/mysql/mysql/time_zone.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/help_relation.MAI to /var/lib/mysql/mysql/help_relation.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/help_keyword.MAI to /var/lib/mysql/mysql/help_keyword.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/func.MAD to /var/lib/mysql/mysql/func.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/servers.MAD to /var/lib/mysql/mysql/servers.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/columns_priv.frm to /var/lib/mysql/mysql/columns_priv.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/tables_priv.MAD to /var/lib/mysql/mysql/tables_priv.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/db.MAI to /var/lib/mysql/mysql/db.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/help_category.frm to /var/lib/mysql/mysql/help_category.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/slow_log.CSM to /var/lib/mysql/mysql/slow_log.CSM
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/time_zone_transition_type.MAI to /var/lib/mysql/mysql/time_zone_transition_type.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/table_stats.frm to /var/lib/mysql/mysql/table_stats.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/proc.MAI to /var/lib/mysql/mysql/proc.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/roles_mapping.MAI to /var/lib/mysql/mysql/roles_mapping.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/proxies_priv.MAI to /var/lib/mysql/mysql/proxies_priv.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/procs_priv.frm to /var/lib/mysql/mysql/procs_priv.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/help_keyword.MAD to /var/lib/mysql/mysql/help_keyword.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/global_priv.frm to /var/lib/mysql/mysql/global_priv.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/plugin.frm to /var/lib/mysql/mysql/plugin.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/general_log.CSV to /var/lib/mysql/mysql/general_log.CSV
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/db.frm to /var/lib/mysql/mysql/db.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/time_zone_transition.MAI to /var/lib/mysql/mysql/time_zone_transition.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/time_zone_leap_second.MAI to /var/lib/mysql/mysql/time_zone_leap_second.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/servers.MAI to /var/lib/mysql/mysql/servers.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/global_priv.MAD to /var/lib/mysql/mysql/global_priv.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/time_zone_leap_second.frm to /var/lib/mysql/mysql/time_zone_leap_second.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/column_stats.MAD to /var/lib/mysql/mysql/column_stats.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/time_zone.frm to /var/lib/mysql/mysql/time_zone.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/help_keyword.frm to /var/lib/mysql/mysql/help_keyword.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/column_stats.MAI to /var/lib/mysql/mysql/column_stats.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/general_log.frm to /var/lib/mysql/mysql/general_log.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/time_zone.MAI to /var/lib/mysql/mysql/time_zone.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/event.MAI to /var/lib/mysql/mysql/event.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/help_topic.MAI to /var/lib/mysql/mysql/help_topic.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/procs_priv.MAI to /var/lib/mysql/mysql/procs_priv.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/general_log.CSM to /var/lib/mysql/mysql/general_log.CSM
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/db.opt to /var/lib/mysql/mysql/db.opt
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/gtid_slave_pos.frm to /var/lib/mysql/mysql/gtid_slave_pos.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/plugin.MAI to /var/lib/mysql/mysql/plugin.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/innodb_index_stats.ibd to /var/lib/mysql/mysql/innodb_index_stats.ibd
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/roles_mapping.frm to /var/lib/mysql/mysql/roles_mapping.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/help_topic.MAD to /var/lib/mysql/mysql/help_topic.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/help_relation.MAD to /var/lib/mysql/mysql/help_relation.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/time_zone_name.frm to /var/lib/mysql/mysql/time_zone_name.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/proxies_priv.frm to /var/lib/mysql/mysql/proxies_priv.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/help_category.MAD to /var/lib/mysql/mysql/help_category.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/servers.frm to /var/lib/mysql/mysql/servers.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/time_zone_transition_type.MAD to /var/lib/mysql/mysql/time_zone_transition_type.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/func.frm to /var/lib/mysql/mysql/func.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/proxies_priv.MAD to /var/lib/mysql/mysql/proxies_priv.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/innodb_table_stats.ibd to /var/lib/mysql/mysql/innodb_table_stats.ibd
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/time_zone_name.MAD to /var/lib/mysql/mysql/time_zone_name.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/roles_mapping.MAD to /var/lib/mysql/mysql/roles_mapping.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/innodb_table_stats.frm to /var/lib/mysql/mysql/innodb_table_stats.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/global_priv.MAI to /var/lib/mysql/mysql/global_priv.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/event.frm to /var/lib/mysql/mysql/event.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/user.frm to /var/lib/mysql/mysql/user.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/columns_priv.MAI to /var/lib/mysql/mysql/columns_priv.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/help_category.MAI to /var/lib/mysql/mysql/help_category.MAI
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/table_stats.MAD to /var/lib/mysql/mysql/table_stats.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/procs_priv.MAD to /var/lib/mysql/mysql/procs_priv.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/time_zone_leap_second.MAD to /var/lib/mysql/mysql/time_zone_leap_second.MAD
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./mysql/index_stats.frm to /var/lib/mysql/mysql/index_stats.frm
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./aria_log_control to /var/lib/mysql/aria_log_control
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./xtrabackup_info to /var/lib/mysql/xtrabackup_info
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./ib_buffer_pool to /var/lib/mysql/ib_buffer_pool
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./db/db.opt to /var/lib/mysql/db/db.opt
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./performance_schema/db.opt to /var/lib/mysql/performance_schema/db.opt
[01] 2021-05-31 18:46:05         ...done
[01] 2021-05-31 18:46:05 Copying ./aria_log.00000001 to /var/lib/mysql/aria_log.00000001
[01] 2021-05-31 18:46:05         ...done
[00] 2021-05-31 18:46:05 completed OK!
Database initialized from /mysqlbase/
+ echo 'Database initialized from /mysqlbase/'
+ rm -f /tmp/initializing
++ cat /var/lib/mysql/db_mariadb_version.txt
+ database_db_version=10.5
+ '[' 10.5 '!=' 10.5 ']'
+ echo 10.5
+ cp -r /home/.my.cnf /home/.bashrc /home/mysql/
cp: target '/home/mysql/' is not a directory
```
</p>
</details>

---

DDEV image
`docker run drud/ddev-dbserver-mariadb-10.5:v1.17.0`
Error:
```
mysqld: Please consult the Knowledge Base to find out how to run mysqld as root! 
```

<details><summary>Full log</summary>
<p>

```
+ set -eu
+ set -o pipefail
+ SOCKET=/var/tmp/mysql.sock
+ rm -f /tmp/healthy
+ '[' 0 = 2 -a '' = restore_snapshot ']'
++ PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/sbin:/usr/local/bin:/usr/local/mysql/bin
++ mysqld -V
++ awk '{sub( /\.[0-9]+(-.*)?$/, "", $3); print $3 }'
+ server_db_version=10.5
++ echo '/mnt/ddev_config/mysql/*.cnf'
+ '[' -d /mnt/ddev_config/mysql -a '/mnt/ddev_config/mysql/*.cnf' '!=' '/mnt/ddev_config/mysql/*.cnf' ']'
+ export BACKUPTOOL=mariabackup
+ BACKUPTOOL=mariabackup
+ command -v xtrabackup
+ '[' '!' -f /var/lib/mysql/db_mariadb_version.txt ']'
+ '[' -z '' ']'
+ touch /tmp/initializing
+ target=/mysqlbase/
++ basename /mysqlbase/
+ name=mysqlbase
+ rm -rf '/var/lib/mysql/*' '/var/lib/mysql/.[a-z]*'
+ mariabackup --prepare --skip-innodb-use-native-aio --target-dir /mysqlbase/ --user=root --password=root --socket=/var/tmp/mysql.sock
+ tee /var/log/mariabackup_prepare_mysqlbase.log
Warning: World-writable config file '/mysqlbase/backup-my.cnf' is ignored
Warning: World-writable config file '/mysqlbase/backup-my.cnf' is ignored
Warning: World-writable config file '/mysqlbase/backup-my.cnf' is ignored
mariabackup based on MariaDB server 10.5.9-MariaDB debian-linux-gnu (aarch64)
[00] 2021-05-31 18:38:16 cd to /mysqlbase/
[00] 2021-05-31 18:38:16 open files limit requested 0, set to 1048576
[00] 2021-05-31 18:38:16 This target seems to be not prepared yet.
[00] 2021-05-31 18:38:16 mariabackup: using the following InnoDB configuration for recovery:
[00] 2021-05-31 18:38:16 innodb_data_home_dir = .
[00] 2021-05-31 18:38:16 innodb_data_file_path = ibdata1:10M:autoextend
[00] 2021-05-31 18:38:16 innodb_log_group_home_dir = .
[00] 2021-05-31 18:38:16 Starting InnoDB instance for recovery.
[00] 2021-05-31 18:38:16 mariabackup: Using 104857600 bytes for buffer pool (set by --use-memory parameter)
2021-05-31 18:38:16 0 [Note] InnoDB: Uses event mutexes
2021-05-31 18:38:16 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2021-05-31 18:38:16 0 [Note] InnoDB: Number of pools: 1
2021-05-31 18:38:16 0 [Note] InnoDB: Using ARMv8 crc32 + pmull instructions
2021-05-31 18:38:16 0 [Note] InnoDB: Initializing buffer pool, total size = 104857600, chunk size = 104857600
2021-05-31 18:38:16 0 [Note] InnoDB: Completed initialization of buffer pool
2021-05-31 18:38:16 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
2021-05-31 18:38:16 0 [Note] InnoDB: Starting crash recovery from checkpoint LSN=45130,45130
[00] 2021-05-31 18:38:16 Last binlog file , position 0
[00] 2021-05-31 18:38:16 completed OK!
+ mariabackup --copy-back --skip-innodb-use-native-aio --force-non-empty-directories --target-dir /mysqlbase/ --user=root --password=root --socket=/var/tmp/mysql.sock
+ tee /var/log/mariabackup_copy_back_mysqlbase.log
mariabackup based on MariaDB server 10.5.9-MariaDB debian-linux-gnu (aarch64)
[01] 2021-05-31 18:38:16 Copying ibdata1 to /var/lib/mysql/ibdata1
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./db_mariadb_version.txt to /var/lib/mysql/db_mariadb_version.txt
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./test/db.opt to /var/lib/mysql/test/db.opt
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/table_stats.MAI to /var/lib/mysql/mysql/table_stats.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/tables_priv.frm to /var/lib/mysql/mysql/tables_priv.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/slow_log.CSV to /var/lib/mysql/mysql/slow_log.CSV
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/db.MAD to /var/lib/mysql/mysql/db.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/slow_log.frm to /var/lib/mysql/mysql/slow_log.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone_transition.frm to /var/lib/mysql/mysql/time_zone_transition.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/proc.MAD to /var/lib/mysql/mysql/proc.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone_transition.MAD to /var/lib/mysql/mysql/time_zone_transition.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/help_relation.frm to /var/lib/mysql/mysql/help_relation.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/tables_priv.MAI to /var/lib/mysql/mysql/tables_priv.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/column_stats.frm to /var/lib/mysql/mysql/column_stats.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone_transition_type.frm to /var/lib/mysql/mysql/time_zone_transition_type.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/help_topic.frm to /var/lib/mysql/mysql/help_topic.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/index_stats.MAD to /var/lib/mysql/mysql/index_stats.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/func.MAI to /var/lib/mysql/mysql/func.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/transaction_registry.ibd to /var/lib/mysql/mysql/transaction_registry.ibd
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/columns_priv.MAD to /var/lib/mysql/mysql/columns_priv.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/innodb_index_stats.frm to /var/lib/mysql/mysql/innodb_index_stats.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/transaction_registry.frm to /var/lib/mysql/mysql/transaction_registry.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone_name.MAI to /var/lib/mysql/mysql/time_zone_name.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/index_stats.MAI to /var/lib/mysql/mysql/index_stats.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/event.MAD to /var/lib/mysql/mysql/event.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/gtid_slave_pos.ibd to /var/lib/mysql/mysql/gtid_slave_pos.ibd
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/plugin.MAD to /var/lib/mysql/mysql/plugin.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/proc.frm to /var/lib/mysql/mysql/proc.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone.MAD to /var/lib/mysql/mysql/time_zone.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/help_relation.MAI to /var/lib/mysql/mysql/help_relation.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/help_keyword.MAI to /var/lib/mysql/mysql/help_keyword.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/func.MAD to /var/lib/mysql/mysql/func.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/servers.MAD to /var/lib/mysql/mysql/servers.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/columns_priv.frm to /var/lib/mysql/mysql/columns_priv.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/tables_priv.MAD to /var/lib/mysql/mysql/tables_priv.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/db.MAI to /var/lib/mysql/mysql/db.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/help_category.frm to /var/lib/mysql/mysql/help_category.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/slow_log.CSM to /var/lib/mysql/mysql/slow_log.CSM
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone_transition_type.MAI to /var/lib/mysql/mysql/time_zone_transition_type.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/table_stats.frm to /var/lib/mysql/mysql/table_stats.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/proc.MAI to /var/lib/mysql/mysql/proc.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/roles_mapping.MAI to /var/lib/mysql/mysql/roles_mapping.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/proxies_priv.MAI to /var/lib/mysql/mysql/proxies_priv.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/procs_priv.frm to /var/lib/mysql/mysql/procs_priv.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/help_keyword.MAD to /var/lib/mysql/mysql/help_keyword.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/global_priv.frm to /var/lib/mysql/mysql/global_priv.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/plugin.frm to /var/lib/mysql/mysql/plugin.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/general_log.CSV to /var/lib/mysql/mysql/general_log.CSV
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/db.frm to /var/lib/mysql/mysql/db.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone_transition.MAI to /var/lib/mysql/mysql/time_zone_transition.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone_leap_second.MAI to /var/lib/mysql/mysql/time_zone_leap_second.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/servers.MAI to /var/lib/mysql/mysql/servers.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/global_priv.MAD to /var/lib/mysql/mysql/global_priv.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone_leap_second.frm to /var/lib/mysql/mysql/time_zone_leap_second.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/column_stats.MAD to /var/lib/mysql/mysql/column_stats.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone.frm to /var/lib/mysql/mysql/time_zone.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/help_keyword.frm to /var/lib/mysql/mysql/help_keyword.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/column_stats.MAI to /var/lib/mysql/mysql/column_stats.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/general_log.frm to /var/lib/mysql/mysql/general_log.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone.MAI to /var/lib/mysql/mysql/time_zone.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/event.MAI to /var/lib/mysql/mysql/event.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/help_topic.MAI to /var/lib/mysql/mysql/help_topic.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/procs_priv.MAI to /var/lib/mysql/mysql/procs_priv.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/general_log.CSM to /var/lib/mysql/mysql/general_log.CSM
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/db.opt to /var/lib/mysql/mysql/db.opt
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/gtid_slave_pos.frm to /var/lib/mysql/mysql/gtid_slave_pos.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/plugin.MAI to /var/lib/mysql/mysql/plugin.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/innodb_index_stats.ibd to /var/lib/mysql/mysql/innodb_index_stats.ibd
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/roles_mapping.frm to /var/lib/mysql/mysql/roles_mapping.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/help_topic.MAD to /var/lib/mysql/mysql/help_topic.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/help_relation.MAD to /var/lib/mysql/mysql/help_relation.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone_name.frm to /var/lib/mysql/mysql/time_zone_name.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/proxies_priv.frm to /var/lib/mysql/mysql/proxies_priv.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/help_category.MAD to /var/lib/mysql/mysql/help_category.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/servers.frm to /var/lib/mysql/mysql/servers.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone_transition_type.MAD to /var/lib/mysql/mysql/time_zone_transition_type.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/func.frm to /var/lib/mysql/mysql/func.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/proxies_priv.MAD to /var/lib/mysql/mysql/proxies_priv.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/innodb_table_stats.ibd to /var/lib/mysql/mysql/innodb_table_stats.ibd
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone_name.MAD to /var/lib/mysql/mysql/time_zone_name.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/roles_mapping.MAD to /var/lib/mysql/mysql/roles_mapping.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/innodb_table_stats.frm to /var/lib/mysql/mysql/innodb_table_stats.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/global_priv.MAI to /var/lib/mysql/mysql/global_priv.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/event.frm to /var/lib/mysql/mysql/event.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/user.frm to /var/lib/mysql/mysql/user.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/columns_priv.MAI to /var/lib/mysql/mysql/columns_priv.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/help_category.MAI to /var/lib/mysql/mysql/help_category.MAI
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/table_stats.MAD to /var/lib/mysql/mysql/table_stats.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/procs_priv.MAD to /var/lib/mysql/mysql/procs_priv.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/time_zone_leap_second.MAD to /var/lib/mysql/mysql/time_zone_leap_second.MAD
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./mysql/index_stats.frm to /var/lib/mysql/mysql/index_stats.frm
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./aria_log_control to /var/lib/mysql/aria_log_control
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./xtrabackup_info to /var/lib/mysql/xtrabackup_info
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./ib_buffer_pool to /var/lib/mysql/ib_buffer_pool
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./db/db.opt to /var/lib/mysql/db/db.opt
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./performance_schema/db.opt to /var/lib/mysql/performance_schema/db.opt
[01] 2021-05-31 18:38:16         ...done
[01] 2021-05-31 18:38:16 Copying ./aria_log.00000001 to /var/lib/mysql/aria_log.00000001
[01] 2021-05-31 18:38:16         ...done
[00] 2021-05-31 18:38:16 completed OK!
Database initialized from /mysqlbase/
+ echo 'Database initialized from /mysqlbase/'
+ rm -f /tmp/initializing
++ cat /var/lib/mysql/db_mariadb_version.txt
+ database_db_version=10.5
+ '[' 10.5 '!=' 10.5 ']'
+ echo 10.5
+ cp -r /home/.my.cnf /home/.bashrc /root/
+ mkdir -p /mnt/ddev-global-cache/bashhistory/20fdd34c34ac
+ echo
+ echo 'MySQL init process done. Ready for start up.'
+ echo
+ echo 'Starting mysqld.'
+ exec mysqld --server-id=0
+ tail -f /var/log/mysqld.log

MySQL init process done. Ready for start up.

Starting mysqld.
2021-05-31 18:38:16 0 [Warning] option 'server_id': unsigned value 0 adjusted to 1
2021-05-31 18:38:16 0 [Note] mysqld (mysqld 10.5.9-MariaDB-1:10.5.9+maria~focal-log) starting as process 1 ...
mysqld: Please consult the Knowledge Base to find out how to run mysqld as root!
2021-05-31 18:38:16 0 [ERROR] Aborting
```
</p>
</details>

rfay commented:
Since you haven't been back to this to manually test or anything, I'm going to close this for now. Happy to reopen when you're ready to work with it some more and go through manual building and testing. As always, happy to continue the conversation here or elsewhere.

hussainweb commented:
I understand. I have been involved with a few other things and will reopen this once I get a chance to move ahead here. Thanks.
